src/histogram_equalization.jl

```diff
@@ -127,6 +132,20 @@ function get_rgb_channels(img)
     return cat(redc, greenc, bluec; dims=3)
 end
 
+"""
+    rgb2gray(rgbchannels::Array{Float64, 3})
+
+Convert an array of RGB channel data to grayscale in the range [0, 255].
+
+Identical to MATLAB `rgb2gray` (https://www.mathworks.com/help/matlab/ref/rgb2gray.html).
+"""
+function rgb2gray(rgbchannels::Array{Float64,3})
+    r, g, b = [to_uint8(rgbchannels[:, :, i]) for i in 1:3]
+    # Reusing the r array to store the equalized gray image
+    r .= to_uint8(0.2989 * r .+ 0.5870 * g .+ 0.1140 * b)
```

Comment:
Why reuse the same array? I assume you're trying to save some memory with this. Does it actually save any memory? Does it create a new array in memory anyway and then assign that to the `r` variable because `r` appears in the right hand side?

Reply:
This approach saves memory in general by preventing another allocation. In this case the RHS array is generated in memory, as you point out, and it's immediately returned so no savings. But it would have yielded savings if RHS would have been used for another operation. Thus, this or `return to_uint8(...)` does the same thing.